homeassistant/components/light/__init__.py

```diff
@@ -207,6 +215,27 @@ def toggle(hass, entity_id=None, transition=None):
     hass.services.call(DOMAIN, SERVICE_TOGGLE, data)
 
 
+def preprocess_turn_on_alternatives(params):
+    """Processing extra data for turn light on request."""
+    profile = Profiles.get(params.pop(ATTR_PROFILE, None))
+    if profile is not None:
+        params.setdefault(ATTR_XY_COLOR, profile[:2])
+        params.setdefault(ATTR_BRIGHTNESS, profile[2])
+
+    color_name = params.pop(ATTR_COLOR_NAME, None)
+    if color_name is not None:
+        params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
```

Comment:
Would it make sense to just mark rgb_color and color_name exclusive in the voluptuous schema? That way they can never be specified together. Now it might be confusing to the user why one parameter would overrule the other.